qiskit/optimization/algorithms/admm_optimizer.py

```diff
@@ -375,6 +375,15 @@ def solve(self, problem: QuadraticProgram) -> ADMMOptimizationResult:
                                         variables=base_result.variables,
                                         state=self._state)
 
+        # convert back integer to binary
+        result = cast(ADMMOptimizationResult, int2bin.decode(result))
+
+        # check for feasibility
+        if QuadraticProgram.is_feasible(problem, result.x):
+            result.status = OptimizationResultStatus.SUCCESS
+        else:
+            result.status = OptimizationResultStatus.INFEASIBLE
```

Comment:
`result.status = ....` should not work/used. An instance of `OptimizationResult` or a sub-class must be immutable.

Reply:
> 
> 
> `result.status = ....` should not work/used. An instance of `OptimizationResult` or a sub-class must be immutable.

I tried and tested this and it is working. So how I go about making it as per your requirements?

Reply:
That's weird if your branch contains the latest changes. `status` is read-only now. So, the only option is to pass `status` to the constructor.

Reply:
There is also result._status, I am guessing that this is read-only and result.status is function used to change the termination status

Reply:
Actually not. 
`_status` sort of a private variable and the `status`property  is now defined as:
```python
    @property
    def status(self) -> OptimizationResultStatus:
        """Returns the termination status of the optimization algorithm.

        Returns:
            The termination status of the algorithm.
        """
        return self._status
```
There's no setter. It was there but not any more. So it is naturally a read only property.

Reply:
Correct, my bad. I will change it so that it is passed as a status flag inside the constructor.